wgpu-hal/src/vulkan/mod.rs

```diff
@@ -240,6 +240,10 @@ pub struct Queue {
     swapchain_fn: khr::Swapchain,
     device: Arc<DeviceShared>,
     family_index: u32,
+    /// This special semaphore is used to synchronize GPU work of
+    /// everything on a queue with... itself. Yes, thanks Vulkan!
```

Comment:
I think this comment could be clearer: it looks like the semaphore is only actually being used to synchronize presentation, whereas other work never waits on it in practice (due to TOP_OF_PIPE).

Reply:
ok, will fix, thank you!